src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java

```diff
@@ -2104,6 +2269,7 @@ protected void track(String eventName, JSONObject properties, boolean isAutomati
     }
 
     private void recordPeopleMessage(JSONObject message) {
+        if (hasOptedOutTracking()) return;
```

Comment:
since `deleteUser` calls `recordPeopleMessage`, have you tested and made sure that by calling `optOutTracking` the profile will actually be removed?

Reply:
since you call `mPersistentIdentity.setOptOutTracking(true);` at the end of `optOutTracking`, it might work. but do you think there's a race condition? if the people queue gets flushed after `mPersistentIdentity.setOptOutTracking(true);` gets called, it won't send out the `deleteUser` event.

Reply:
nope, there is not such case. the analytics worker executes one message at a time.

Reply:
i'm not sure if i follow. i think `deleteUser` enqueues a people message but doesn't flush the queue. if the queue gets flushed after `mPersistentIdentity.setOptOutTracking(true)` getting called, the message won't be sent.

Reply:
We don't have any checks on the `flush` method, so the message will be sent :)